clouddriver-lambda/src/main/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/CreateLambdaAtomicOperation.java

```diff
@@ -69,9 +74,97 @@ private CreateFunctionResult createFunction() {
     request.setCode(code);
     request.setTags(objTag);
 
+    Map<String, String> envVariables = description.getEnvVariables();
+    if (null != envVariables) {
+      request.setEnvironment(new Environment().withVariables(envVariables));
+    }
+
+    if (null != description.getSecurityGroupIds() || null != description.getSubnetIds()) {
+      request.setVpcConfig(
+          new VpcConfig()
+              .withSecurityGroupIds(description.getSecurityGroupIds())
+              .withSubnetIds(description.getSubnetIds()));
+    }
+    request.setDeadLetterConfig(description.getDeadLetterConfig());
+    request.setKMSKeyArn(description.getEncryKMSKeyArn());
+    request.setTracingConfig(description.getTracingConfig());
+
     CreateFunctionResult result = client.createFunction(request);
     updateTaskStatus("Finished Creation of AWS Lambda Function Operation...");
+    if (description.getTargetGroup() != null && !description.getTargetGroup().isEmpty()) {
+
+      updateTaskStatus(
+          String.format(
+              "Started registering lambda to targetGroup (%s)", description.getTargetGroup()));
+      String functionArn = result.getFunctionArn();
+      registerTargetGroup(functionArn, client);
+    }
 
     return result;
   }
+
+  protected String combineAppDetail(String appName, String functionName) {
+    return Names.parseName(functionName).getApp().equals(appName)
```

Comment:
if functionName contains any characters that are invalid for a Frigga name, parseName will return null - it would be worth guarding against that unless the valid set of characters for a function name conforms to the Frigga naming constraints

Reply:
function name confirms to the Frigga naming constraints, But it's good to have a null check. Implemented.